WordPress/gutenberg
# Query block: move patterns modal to dropdown on block toolbar

## What?

Resolves https://github.com/WordPress/gutenberg/issues/63497

This PR:
- replaces the "Replace" button in the block toolbar with a "Change design" button. See: https://github.com/WordPress/gutenberg/pull/66993#issuecomment-2495499304
- and then displays patterns selection content in a drop down


## How?

Extracts the belly from the modal to make it reusable.
Also abstracts the `hasPatterns` test for consistency.

## Testing Instructions

1. Fire up the site editor, and open a template with an existing query block, or create one yourself! TT5 theme is a worthy choice.
2. Then select the query block. Check out the block toolbar. Do you see the "Change design" button? Click on it to display the patterns. 
3. Replace a pattern, make sure that works. 
4. Now narrow the viewport width and ensure it's usable in mobile view.
5. Thank you.


## Screenshots or screencast 




https://github.com/user-attachments/assets/dd9b19d5-b56c-4723-a844-194ae98c14eb
